react/features/app/actions.js

```diff
@@ -147,15 +147,15 @@ function _navigate(state) {
  */
 function _setRoomAndNavigate(newRoom) {
     return (dispatch, getState) => {
-        const oldRoom = getState()['features/base/conference'].room;
-
         dispatch(setRoom(newRoom));
 
         const state = getState();
-        const room = state['features/base/conference'].room;
+        const { app } = state['features/app'];
```

Comment:
The source code from this line to the end of the method would probably better fit in `App`'s `_navigate` method.